ci: retry gke cluster scale up, don't clear cluster at start

nebril opened the pull request:
This step sometimes fails due to another operation (possible started by
gke) working on a cluster, this will allow us to wait through this
operation and properly scale cluster up.

nebril committed:
ci: retry gke cluster scale up

This step sometimes fails due to another operation (possible started by
gke) working on a cluster, this will allow us to wait through this
operation and properly scale cluster up.

Signed-off-by: Maciej Kwiek <maciej@isovalent.com>

nbusseneau commented on test/gke/resize-cluster.sh:
```diff
@@ -39,4 +39,15 @@ if [ "${#node_pools[@]}" -ne 1 ] ; then
+	((resize_retries++))
+done
+
+exit $scaled
```
I was thinking about that last week, but is there any reason most of the pipelines' work is done in `.sh` and not in the pipelines themselves?

If not, I think it could be good to think about moving intelligence out of the scripts and into Jenkinsfiles when appropriate. For example in this case we could have used Jenkins retry blocks rather than a custom bash retry, which I feel is more flexible.

What do you think?

nebril replied:
I would say that the reason for that is that most people contributing are much more comfortable with bash than with jenkinsfiles.

We can definitely talk about moving all this logic to jenkinsfiles, but this script is called from within another script that is already wrapped in retries.

After cluster is resized there are still some other things to be done, so we would need to have yet another script in gke dir (which might be a good idea).

nebril committed:
ci: don't clean selected cluster

We are using freshly created clusters for gke runs, there is no need to
clean it in select-cluster script.

Signed-off-by: Maciej Kwiek <maciej@isovalent.com>
